src/app.ts

```diff
@@ -213,56 +308,193 @@ export async function build(opts = {}) {
     }
   )
 
   server.post(
-    '/issue/credentials',
-    {
-      config: {
-        rawBody: true,
-      },
-      preValidation: [
-        verifyRequestDigest,
-        verifyRequestSignature
-      ]
-    },
-    async (request, reply) => {
-      const req: any = request.body;
-      const credential = req.credential;
-      const options = req.options;
+    '/issue/credentials', async (request, reply) => {
+      const req = request.body as { credential: Credential; options: Object; };
 
-      const result = await sign(credential, options);
-      reply
-        .code(201)
-        .header('Content-Type', 'application/json; charset=utf-8')
-        .send(result);
+      try {
+        // Attach status to credential
+        const { credential, newList } = await credStatusClient.embedCredentialStatus({ credential: req.credential });
+
+        // Setup data necessary for composing signed status credential
+        const issuerDid = publicDids[0].id;
+        const verificationMethod = ensureId(publicDids[0].assertionMethod[0]);
+
+        // Create new status credential only if a new list was created
+        if (newList) {
+          // Create and sign status credential
+          const statusCredentialId = `${credentialStatusUrl}/${newList}`;
+          const statusCredentialDataUnsigned = await composeStatusCredential({ issuerDid, credentialId: statusCredentialId });
+          const statusCredentialData = await sign(statusCredentialDataUnsigned, { verificationMethod });
+
+          // Create and persist status data
+          await credStatusClient.createStatusData(statusCredentialData);
+        }
+
+        // Sign credential
+        const options = req.options;
+        const result = await sign(credential, options);
+
+        // Add new entry to status log
+        const { id: credentialStatusId, statusListCredential, statusListIndex } = credential.credentialStatus;
+        const statusListId = statusListCredential.split('/').slice(-1).pop(); // retrieve status list id from status credential url
+        const statusLogEntry: CredentialStatusLogEntry = {
+          timestamp: (new Date()).toISOString(),
+          credentialId: credential.id || credentialStatusId,
+          credentialIssuer: issuerDid,
+          credentialSubject: credential.credentialSubject?.id,
+          credentialState: CredentialState.Issued,
+          verificationMethod,
+          statusListId,
+          statusListIndex
+        };
+        const statusLogData = await credStatusClient.readLogData();
+        statusLogData.push(statusLogEntry);
+        await credStatusClient.updateLogData(statusLogData);
+
+        reply
+          .code(201)
+          .header('Content-Type', 'application/json; charset=utf-8')
+          .send(result);
+      } catch (error: any) {
+        reply
+          .code(400)
+          .send({ message: error.message });
+      }
     }
   )
 
+  server.post(
+    '/credentials/status', async (request, reply) => {
```

Comment:
**IMPORTANT:** Although we are protecting access between the `sign-and-verify` server and the managed source control resources, we are not protecting access between the client and the server. Should we require the client to use the OAUTH token in the request?

Reply:
This has been resolved in a recent commit.